Sequence page

mantecajc opened the pull request:
Dans cette branche, on a développé les pages séquence, séance, activité et atelier. 

Il n'est pas exactement le même que celui envoyé à PAD-EIG. Certains bugs ont été déjà corrigés ici.

mantecajc committed:
Redesign section page

- display section name in page header
- new design for activity cards
- small changes for  navigation between sections

mantecajc committed:
Redesign workshop page

denier-epnak commented on theme/padplus/classes/output/core_renderer.php:
```diff
@@ -359,6 +386,75 @@ public function context_header_settings_menu() {
+     *
+     * @return string
+     */
+    public function region_main_settings_menu() {
```
We already have an overridden version in coursecat_settings_menu_padplus below. Need discussion

denier-epnak replied:
Simon

denier-epnak commented on theme/padplus/javascript/add-folder-icon.js:
```diff
@@ -0,0 +1,14 @@
+// Allows us to add the folder icon to the page title when we are in a section/activity page.
+var sectionPageClass = document.querySelectorAll('.format-topics');
+
+if (sectionPageClass.length > 0) { // We check if the class is empty or not. It allows us to exclude workshop page.
```
is this essential?

denier-epnak replied:
Javier

mantecajc replied:
done

denier-epnak commented on theme/padplus/lang/en/theme_padplus.php:
```diff
@@ -83,11 +83,22 @@
-$string['btn-coursebox'] = 'View this course';
+$string['activity-session'] = '{$a->total} activity in this session:';
+$string['activities-session'] = '{$a->total} activities in this session:';
+$string['btn-coursebox'] = 'View this cours';
```
typo 'course'

denier-epnak replied:
Javier

denier-epnak commented on theme/padplus/layout/columns2.php:
```diff
@@ -121,15 +121,31 @@
+// Enable divider on coursehome since it comes after the main menu now & set a common text.
 $coursehome = $padnav->get('coursehome');
 if (is_object($coursehome)) {
+    if (course_is_workshop($this->page->course)) {
```
Discussion : we have two different tests for workshop

denier-epnak replied:
Simon

denier-epnak commented on theme/padplus/layout/columns2.php:
```diff
@@ -121,15 +121,31 @@
 $coursehome = $padnav->get('coursehome');
 if (is_object($coursehome)) {
+    if (course_is_workshop($this->page->course)) {
+        $nav->get('coursehome')->text = get_string('sidebar-sumamry-workshop', 'theme_padplus');
```
In such cases, prefer ternary expressions (less indent = easier to read) :

$labelkey = course_is_workshop($this->page->course) ? 'sidebar-sumamry-workshop' : 'sidebar-summary-course';
$coursehome->text = get_string($labelkey, 'theme_padplus');

denier-epnak replied:
Simon

denier-epnak commented on theme/padplus/lang/fr/theme_padplus.php:
```diff
@@ -83,11 +83,22 @@
 $string['settings-category'] = 'Gestion de cette catégorie';
+$string['settings-activity'] = "Paramètres de l'activité";
+$string['sidebar-summary-course'] = 'Dans cette séquence';
+$string['sidebar-sumamry-workshop'] = 'Dans cet atelier';
```
typo summary

denier-epnak replied:
javier

denier-epnak replied:
simon

denier-epnak commented on theme/padplus/layout/columns2.php:
```diff
@@ -121,15 +121,31 @@
 }
 
+// Unlink icons for grades & participants.
+$grades = $nav->get('grades');
```
foreach(array('grades', 'participants') as $item) {
$navitem = $padnav->get($item);
if (is_object($navitem)) {
    $navitem->icon->pix = null;
}
}

denier-epnak replied:
simon

denier-epnak commented on theme/padplus/scss/post/layouts/_sidebar.scss:
```diff
@@ -44,6 +44,12 @@
+    [data-key = 'participants'],
+    [data-key = 'grades'] {
+        span {
+            margin-left: 29px;
```
isn't there some kind of indent stuff built-in Moodle flat navigation? I remember something like this

denier-epnak replied:
simon

denier-epnak replied:
see indent/get_indent

denier-epnak commented on theme/padplus/scss/post/layouts/_sidebar.scss:
```diff
@@ -59,3 +65,12 @@
+}
+
+.fa.fa-graduation-cap:before {
+    content: '\f07c' !important; // icon folder-open fontawesome
```
use theme icon system for overidding icons

denier-epnak replied:
javier

denier-epnak commented on theme/padplus/classes/output/format_topics_renderer.php:
```diff
@@ -0,0 +1,300 @@
+        // Output section activities summary.
+        $o = '';
+        $o .= html_writer::start_tag('div', array('class' => 'section-summary-activities pr-2 mdl-right'));
+        if ($total == 1) {
```
use ternary expression:
$label = ($total === 1) ? 'activity-session' : 'activities-session';

denier-epnak replied:
javier

denier-epnak commented on theme/padplus/classes/output/core/course_renderer.php:
```diff
@@ -420,4 +421,167 @@ public function course_info_box(stdClass $course) {
+            $activitydates = \core\activity_dates::get_dates_for_module($mod, $USER->id);
+        }
+
+        // If there is content AND a link, then display the content here
```
Missing PADPLUS marker

denier-epnak replied:
Javier

denier-epnak commented on theme/padplus/classes/output/core_renderer.php:
```diff
@@ -249,7 +249,20 @@ private function is_dashboard_page() {
+        $course = $this->page->course;
+        $pageurl = $this->page->url->__toString();
+        // We check if we are on the activity page of a course but we exclude workshop activity page.
+        return ((strpos($pageurl, '/mod/') == true) && (course_is_workshop($course) == false));
```
it's a specific case: "mod/lesson" activity

denier-epnak replied:
Wait for discussion

denier-epnak commented on theme/padplus/scss/post/components/_badges.scss:
```diff
@@ -16,7 +17,29 @@
+    }
+}
+
+.badge-success {
```
regression? &.badge-pad-workshop is not linked to badge-pad but to badge-success

denier-epnak replied:
javier

mantecajc replied:
J'en ai profité pour corriger le bug des margins (https://3.basecamp.com/5191032/buckets/24030658/todos/4739288153) et centrer verticalement le texte

denier-epnak commented on theme/padplus/lang/en/theme_padplus.php:
```diff
@@ -83,11 +83,22 @@
+$string['progress-session'] = 'Progress:';
 $string['settings-category'] = 'Management of this category';
+$string['settings-activity'] = "Activity settings";
+$string['sidebar-summary-course'] = 'In this cours';
```
typo 'course'

denier-epnak replied:
javier

denier-epnak commented on theme/padplus/scss/post/pages/_categories.scss:
```diff
@@ -32,11 +32,11 @@
             border: 1px solid theme-color("pad-border");
             border-radius: 8px;
             display: flex;
-            height: 64px;
```
Extract from PR

denier-epnak replied:
javier

mantecajc committed:
Small adjustments for course navigation in sidebar

mantecajc committed:
override: core_renderer method for edit_button

mantecajc committed:
override: format_topics_renderer for section page and activity summary

mantecajc committed:
override: renderer and templates for course/section/completion

mantecajc committed:
Highlight "Quit edition mode" button in red

mantecajc committed:
Redesign course main page

- overall page design
- new design for session cards with activities and progression

mantecajc committed:
override: activity navigation

denier-epnak committed:
override: core_renderer region_main_settings_menu

mantecajc committed:
Redesign activity page

- display section name as page header
- page design
- bottom navigation between activities

denier-epnak committed:
Redesign actions dropdown for categories, participants, activities

- reuse the generic region_main_settings_menu with a custom trigger label
- refactor css to use a common dropdown mixin
- keep the variant location on category page, so that we do not display the dropdown for contributors, only for admins/managers
- remove previous custom coursecat_settings_menu_padplus method
